tests/extern-scan.pl

```diff
@@ -28,18 +28,61 @@
 use strict;
 use warnings;
 
+my $allheaders = 0;
+my $sort = 0;
+
 # we may get the dir root pointed out
-my $root=$ARGV[0] || ".";
+my $root = shift @ARGV;
+while(defined $root) {
+
+    if($root =~ /--heading=(.*)/) {
+        print "$1\n";
+        $root = shift @ARGV;
+        next;
+    }
+    elsif($root =~ /--allheaders/) {
+        $allheaders = 1;
+        $root = shift @ARGV;
+        next;
+    }
+    elsif($root =~ /--sort/) {
+        $sort = 1;
+        $root = shift @ARGV;
+        next;
+    }
+
+    last;
+}
+
+if(!defined $root) {
+    $root=".";
+}
+
+my @incs;
+if($allheaders == 1) {
+    $root = "$root/include/curl";
+    opendir(D, "$root") || die "Cannot open directory $root: $!\n";
+    my @dir = readdir(D);
+    closedir(D);
 
-my @incs = (
-    "$root/include/curl/curl.h",
-    "$root/include/curl/easy.h",
-    "$root/include/curl/mprintf.h",
-    "$root/include/curl/multi.h",
-    "$root/include/curl/urlapi.h",
-    "$root/include/curl/options.h",
-    "$root/include/curl/header.h",
-    );
+    foreach (sort(@dir)) {
+        if($_ =~ /\.h$/) {
+            push(@incs, "$root/$_");
+        }
+    }
+}
+else {
+    @incs = (
+        "$root/include/curl/curl.h",
+        "$root/include/curl/easy.h",
+        "$root/include/curl/mprintf.h",
+        "$root/include/curl/multi.h",
+        "$root/include/curl/urlapi.h",
+        "$root/include/curl/options.h",
+        "$root/include/curl/header.h",
+        "$root/include/curl/websockets.h",
+        );
+}
```

Comment:
Is there actually a use for the not-allheaders case?

Reply:
I didn't fully grasp the original intent, so I left it there, assuming its purpose is to somehow enforce an "append-only" growth of the list for VMS and OS/400 (based on the comment in `test1135`). If I misunderstood or that's not important, I agree we are better off deleting this case.

Reply:
I'm about deleting it, and that will change the order of expected results (because the manual header list was not alpha-sorted). If that's okay, it's ready for commit.

Reply:
> assuming its purpose is to somehow enforce an "append-only" growth

Oh right, I forgot about that and I believe I had that in mind. I believe that is unnecessary because as you say, this list is just reading the headers in some specific order, it does not actually define in which order those builds use these symbols so any such append-only order would have to be defined elsewhere.